hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecords.java

```diff
@@ -45,7 +45,7 @@ public static <T> BufferedRecord<T> fromHoodieRecord(HoodieRecord record, Schema
     String recordKey = hoodieKey == null ? recordContext.getRecordKey(data, schema) : hoodieKey.getRecordKey();
     Integer schemaId = recordContext.encodeAvroSchema(schema);
     Comparable orderingValue = record.getOrderingValue(schema, props, orderingFields);
-    return new BufferedRecord<>(recordKey, recordContext.convertOrderingValueToEngineType(orderingValue), data, schemaId, inferOperation(isDelete, record.getOperation()));
```

Comment:
This seems wrong, the ordering value needs to be in the engine specific type but we're removing that here. Is it being added elsewhere?